### ISSUE TITLE
Is there any easy way to use TeXstudio via TeX Live 2019 in WSL

### ISSUE THREAD

**Issue by OsbertWang** [texstudio-org/texstudio#698]
### Environment
- TeXstudio: 2.12.16
- Qt: 5.12.1
- OS: Windows 10
- TeX distribution: TeX Live 2019 in WSL

### Expected behavior
Now I have installed TeX Live 2019 in WSL.
I want to use TeXstudio in Windows 10 and compile files via TeX Live 2019 in WSL.

### Actual behavior
Now I can define user command to compile files via bash.exe, e.g., "C:\Windows\WinSxS\...\bash.exe" -i -c "?me)".
However, I cannot do forward and inverse search.
I notice that the reason is the path in synctex.

### How to reproduce
Can TeXstudio.exe resolute the path in synctex? Or, does TeXstudio want to develop a WSL edition?

**Comment by ludwigbald** [comment]
I don't think this is a typical usecase. Why don't you simply install TexLive for Windows if you intend to use it from Windows?

**Comment by OsbertWang** [comment]
> I don't think this is a typical usecase. Why don't you simply install TexLive for Windows if you intend to use it from Windows?

I used texlive installed in WSL just for fun at first. Then I find that, for the same file,  xelatex compiles faster in WSL than that in Windows 10, especially for Chinese character. I think it's the reason that I prefer to compiling in WSL rather than in Windows 10. 

I notice that an increasing number of programmers use WSL. The VSCode editor have already support the remote-WSL plug, and The LaTeX Workshop works well with the extension, see [LaTeX Workshop FAQ](https://github.com/James-Yu/LaTeX-Workshop/wiki/FAQ#using-latex-workshop-with-wsl).

However, TeXstudio is the editor I use most, so I really wonder whether there is any easy way to use it via TeX Live in WSL.

**Comment by albspe** [comment]
I have been able to use TexStudio with TexLive on WSL with Ubuntu 18.04 with no issues. The only strange thing is that on one machine I get a high CPU usage compared to the other one. Configuration wise they seem to be both fairly close, so I am not sure what is causing the problem. I am running 2.12.18 version on both machines.

**Comment by sierxue** [comment]
如果电脑里面没有 aer.bst 文件的话，文献出不来，好像很难查找原因。因为 aer.bst 在宏包 economic 里面，有没有可能 require 这个宏包，这样容易查找出错误？

**Comment by K-miy** [comment]
https://marcus.handte.org/2017/11/18/using-texstudio-with-texlive-from-ubuntu-on-wsl/

**Comment by butcher211** [comment]
Hello, I would recommend to use a docker image (i. e. Island of Tex) and using commands like `wsl docker run -i --rm --name latex -v "$PWD":/usr/src/app -w /usr/src/app registry.gitlab.com/islandoftex/images/texlive:latest latexmk -silent -synctex=1 %.tex` in Configuration > Commands. It was just an idea in the first place but now this is my daily workflow building tex files.

**Comment by dbitouze** [comment]
I second the suggestion of Semih, except I recommend the docker image:

`registry.gitlab.com/islandoftex/images/texlive:latest-with-cache`

which provides more tools (though I don't remember which ones :$).

Two remarks:

1. The compilation of `.tex` files is almost 2 times faster with the docker image than with the ordinary (vanilla installed) TeX Live on my Linux box which has the following characteristics:
    - Operating System: Mageia 8
    - KDE Plasma Version: 5.20.4
    - KDE Frameworks Version: 5.76.0
    - Qt Version: 5.15.2
    - Kernel Version: 5.15.11-desktop-3.mga8
    - OS Type: 64-bit
    - Processors: 4 × Intel® Core™ i5-7500 CPU @ 3.40GHz
    - Memory: 15.5 Gio of RAM
    - Graphics Processor: Mesa Intel® HD Graphics 630.
1. A drawback is that one can't use (at least I don't see how) the TEXMFHOME personal TDS tree.

**Comment by muzimuzhi** [comment]
> I second the suggestion of Semih, except I recommend the docker image:
> 
> `registry.gitlab.com/islandoftex/images/texlive:latest-with-cache`
> 
> which provides more tools (though I don't remember which ones :$).

Out of curiosity, I searched a bit and here's what I find:

Currently `latest-with-cache` comes with pre-generated LuaTeX font cache & ConTeXt file database cache. See 
 - [islandoftex/images/texlive#3](https://gitlab.com/islandoftex/images/texlive/-/issues/3) "Would it be possible to pre-generate the LuaTeX fontcache?"
 - [`Dockerfile.latest-cache`](https://gitlab.com/islandoftex/images/texlive/-/blob/master/Dockerfile.latest-cache)

For texlive2022, the cache will be generated for `latest` image by default, and the `*-with-cache` images will retire. See [islandoftex/images/texlive!11](https://gitlab.com/islandoftex/images/texlive/-/merge_requests/11) (the 11th merge request).

**Comment by Joan-Hou** [comment]
> https://marcus.handte.org/2017/11/18/using-texstudio-with-texlive-from-ubuntu-on-wsl/
> 
> Used that and it works so far :)

this is amazing! i have tried several ways to compile my.tex in texStudio by the TexLive installed in WSL, only this one works so far. Since some configuration commands is not very clear in the pic of the link, for other user's convenience, I pasted the configuration commands from the picture in the link for reference. Feel free to copy it yourself and try it! And please substitude with your own output under `C:\Windows\WinSxS`.
For LaTex: `C:\Windows\WinSxS\amd64_microsoft-windows-lxss-bash_31bf3856ad364e35_10.0.19041.4474_none_b44cc5c971d28390\bash.exe -c "latex -src -interaction=nonstopmode ?me"`
For PdfLaTex: `C:\Windows\WinSxS\amd64_microsoft-windows-lxss-bash_31bf3856ad364e35_10.0.19041.4474_none_b44cc5c971d28390\bash.exe -c "pdflatex -synctex=1 -interaction=nonstopmode ?me"`
For DviPs: `C:\Windows\WinSxS\amd64_microsoft-windows-lxss-bash_31bf3856ad364e35_10.0.19041.4474_none_b44cc5c971d28390\bash.exe -c "dvips -o %.ps %.dvi"`
For DviPng: 
`C:\Windows\WinSxS\amd64_microsoft-windows-lxss-bash_31bf3856ad364e35_10.0.19041.4474_none_b44cc5c971d28390\bash.exe -c "dvipng -T tight -D 120 %.dvi"`
For Ps2Pdf: `C:\Windows\WinSxS\amd64_microsoft-windows-lxss-bash_31bf3856ad364e35_10.0.19041.4474_none_b44cc5c971d28390\bash.exe -c "ps2pdf ?m.ps"  `
The others, please type yourself. :)

**Comment by OsbertWang** [comment]
> this is amazing! i have tried several ways to compile my.tex in texStudio by the TexLive installed in WSL, only this one works so far. Since some configuration commands is not very clear in the pic of the link, for other user's convenience, I pasted the configuration commands from the picture in the link for reference. Feel free to copy it yourself and try it! And please substitude with your own output under `C:\Windows\WinSxS`. For LaTex: `C:\Windows\WinSxS\amd64_microsoft-windows-lxss-bash_31bf3856ad364e35_10.0.19041.4474_none_b44cc5c971d28390\bash.exe -c "latex -src -interaction=nonstopmode ?me"` For PdfLaTex: `C:\Windows\WinSxS\amd64_microsoft-windows-lxss-bash_31bf3856ad364e35_10.0.19041.4474_none_b44cc5c971d28390\bash.exe -c "pdflatex -synctex=1 -interaction=nonstopmode ?me"` For DviPs: `C:\Windows\WinSxS\amd64_microsoft-windows-lxss-bash_31bf3856ad364e35_10.0.19041.4474_none_b44cc5c971d28390\bash.exe -c "dvips -o %.ps %.dvi"` For DviPng: `C:\Windows\WinSxS\amd64_microsoft-windows-lxss-bash_31bf3856ad364e35_10.0.19041.4474_none_b44cc5c971d28390\bash.exe -c "dvipng -T tight -D 120 %.dvi"` For Ps2Pdf: `C:\Windows\WinSxS\amd64_microsoft-windows-lxss-bash_31bf3856ad364e35_10.0.19041.4474_none_b44cc5c971d28390\bash.exe -c "ps2pdf ?m.ps" ` The others, please type yourself. :)

Have you tried executing the following command to add a new icon?
```
WSL-latexmk: powershell.exe -Command Start-Process "wsl.exe" -ArgumentList "latexmk -pdfxe -halt-on-error -interaction=nonstopmode -file-line-error -silent -synctex=1 $(wslpath -u '%')" -Wait

```
I also recall that when using `texdoc` in WSL, I modified `texdoc.cnf` as follows to enable the correct viewer:
```
viewer_pdf = (powershell.exe -Command Start-Process $(wslpath -w %s)) &
```
Perhaps we can follow a similar approach for your use case.
